mangarack-component-core/src/providers/dynastyreader/default.ts

```diff
@@ -0,0 +1,35 @@
+import * as mio from '../../default';
+import {createSeriesAsync} from './series';
+
+/**
+ * Represents the provider.
+ * @internal
+ */
+export const dynastyreader: mio.IProvider = {
+  /**
+   * Contains the name.
+   */
+  name: 'dynastyreader',
+
+  /**
+   * Determines whether the address is a supported address.
+   * @param address The address.
+   * @return Indicates whether the address is a supported address.
+   */
+  isSupported: function(address: string): boolean {
+    return /^http:\/\/dynasty-scans\.com\/series\//.test(address) && !/\/$/.test(address);
+  },
+
+  /**
+   * Promises the series.
+   * @param address The address.
+   * @return The promise for the series.
+   */
+  seriesAsync: function(address: string): Promise<mio.ISeries> {
+    if (!dynastyreader.isSupported(address)) {
+      throw new Error(`Invalid series address: ${address}`);
+    }
+
```

Comment:
A `throw` or a `return` will always stop and exit the function, so once a `throw` is entered, it's implicit that it will exit the function. It is also common practice to guard classes (conditionals) so that the function can exit without going further, as it has been done in other areas of the `mangarack` code. I'm not sure putting an `else` actually improves readability; because this can lead to "conditional hell", but it is really preference so I'll respect whatever code style you prefer.